apps/gallery/js/info.js

```diff
@@ -44,18 +52,86 @@ function showFileInformation(fileinfo) {
           args: args
         },
         'info-type': {raw: fileinfo.type},
-        'info-date': {raw: MediaUtils.formatDate(fileinfo.date)},
+        'info-date': {raw: formattedDate},
         'info-resolution': {
           raw: fileinfo.metadata.width + 'x' + fileinfo.metadata.height
-        }
+        },
       };
+      // placeholders for Exif. Also hide the exif fields.
+      exifFields.forEach((elem) => {
+        data[elem] = {raw: ''};
+        setFieldVisibility(elem, false);
+      });
 
       // Populate info overlay view
       MediaUtils.populateMediaInfo(data);
+
+      LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+        photodb.getFile(fileinfo.name, (imagefile) => {
+          JPEGParser.readExifMetaData(imagefile, (error, metadata) => {
+            if (!metadata) {
+              return;
+            }
+
+            var data = {};
+            // We don't seem to supply FNumber
+            var fnumber;
+            if (metadata.FNumber && metadata.FNumber.denominator) {
+              fnumber = metadata.FNumber.numerator /
+                metadata.FNumber.denominator;
+            } else if (metadata.ApertureValue &&
+                       metadata.ApertureValue.denominator) {
+              fnumber = Math.pow(2, (metadata.ApertureValue.numerator /
+                                     metadata.ApertureValue.denominator) / 2);
+            }
+            var exposureTime = metadata.ExposureTime ?
+                MediaUtils.formatExifExposure(metadata.ExposureTime) : null;
+            var focalLength = metadata.FocalLength &&
+                metadata.FocalLength.denominator ?
+                (metadata.FocalLength.numerator /
+                 metadata.FocalLength.denominator) + 'mm' : null;
+            // use Exif date if available
+            if (metadata.DateTimeOriginal) {
+              formattedDate = MediaUtils.formatExifDate(
+                metadata.DateTimeOriginal);
+            }
+            data['info-date'] = { raw: formattedDate };
```

Comment:
Just to be safer, may be we should check for exif formattedDate for undefined before setting data['info-date']